addons/addon-base-raas/packages/base-raas-services/lib/alb/alb-service.js

```diff
@@ -338,6 +337,80 @@ class ALBService extends Service {
     // return auditWriterService.write(requestContext, auditEvent);
     return auditWriterService.writeAndForget(requestContext, auditEvent);
   }
+
+  /**
+   * Method to modify rule. The method modify the rule using the ALB SDK client.
+   * Tags are read form the resolvedVars so the billing will happen properly
+   *
+   * @param requestContext
+   * @param resolvedVars
+   * @returns {Promise<D & {$response: Response<D, E>}>}
+   */
+  async modifyRule(requestContext, resolvedVars) {
```

Comment:
Can concurrent requests by say the workspace owner (researcher) and an admin cause one of the requests to be omitted. If yes, can we add some locking mechanism to ensure that concurrent requests don't overwrite the rules.

Reply:
modifyRule function declaration as included into the existing locking mechanism. Please review